src/main/java/frc/robot/subsystems/Turret.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.revrobotics.spark.ClosedLoopSlot;
+import com.revrobotics.spark.SparkBase;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.config.ClosedLoopConfig.FeedbackSensor;
+import com.revrobotics.spark.config.SparkBaseConfig.IdleMode;
+import com.revrobotics.spark.config.SparkMaxConfig;
+
+import edu.wpi.first.math.trajectory.TrapezoidProfile.State;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class Turret extends SubsystemBase {
+
+  // Constants to limit the hood rotation speed
+  // max vel: 1 rotation = 10 seconds and then gear_ratio
+  private static final double MAX_VEL_ROT_PER_SEC = 4.0;
```

Comment:
Please set a sensible max velocity. 4 rps is 240 RPM. That is ridiculous for a turret. Even if this cannot be reached because of the acceleration and/or physical limits, we should be setting numbers in a sensible range. Think of it like current limits: it is a backstop against something crazy happening, or someone copying the code to a different system where it is a disaster.